### PR TITLE
fix dag deploy updating on accident

### PR DESCRIPTION
**Pull request by sunkickr**
## Description

When running the command `astro deployment update` dag deploy is always turned off. This is because the case for dag deploy not changing didn't exist.

Added an assert to the update test to test for this bug.

Related #XXX

## 🧪 Functional Testing

> List the functional testing steps to confirm this feature or fix.

## 📸 Screenshots
screenshot of updating the description of both a deployment with dag deploy enabled and disabled. Showing that in both cases dag deploy stays the same:
Screenshot 2024-02-02 at 11 46 31 AM

## 📋 Checklist

- [ ] Rebased from the main (or release if patching) branch (before testing)
- [ ] Ran `make test` before taking out of draft
- [ ] Ran `make lint` before taking out of draft
- [ ] Added/updated applicable tests
- [ ] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [ ] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cloud/deployment/deployment.go [hunk 1]
```diff
@@ -69,6 +69,7 @@ var (
 	tickNum                    = 10
 	timeoutNum                 = 180
 	dedicatedDeploymentRequest = astroplatformcore.UpdateDedicatedDeploymentRequest{}
+	dagDeployEnabled           bool
```
Comment (vandyliu) [inline cloud/deployment/deployment.go:72]: what was the reasoning behind putting it up here, it seems to fit the style of code the way we had it before

File: cloud/deployment/deployment.go [hunk 2]
```diff
@@ -768,6 +767,10 @@ func Update(deploymentID, name, ws, description, deploymentName, dagDeploy, exec
 		dagDeployEnabled = false
 	}
 
+	if dagDeploy == "" {
```
Comment (vandyliu) [inline cloud/deployment/deployment.go:770]:
```suggestion
	else if dagDeploy == "" {
```

**Comment by kushalmalani** [comment]
Can you add some screenshots for this. This bug fix is critical

File: cloud/deployment/deployment_test.go [hunk 3]
```diff
@@ -1288,6 +1288,7 @@ func TestUpdate(t *testing.T) { //nolint
 		// success with hybrid type
 		err := Update("", "", ws, "", "", "", CeleryExecutor, "", "", "", "", "", "", "", 0, 0, workerQueueRequest, hybridQueueList, newEnvironmentVariables, true, mockCoreClient, mockPlatformCoreClient)
```
Comment (kushalmalani) [inline cloud/deployment/deployment_test.go:1289]: Can we add a comment up here w.r.t to what are we updating? 

Eventually would want to get rid of that huge list of arguments and have defined input type, but we can do that later.

File: cloud/deployment/deployment.go (hunk 1 above)
Comment (sunkickr) [reply cloud/deployment/deployment.go:72]: We need it exist outside of the function so we can assert that it is the right value in the test

File: cloud/deployment/deployment.go (hunk 2 above)
Comment (sunkickr) [reply cloud/deployment/deployment.go]: fixed

File: cloud/deployment/deployment_test.go (hunk 3 above)
Comment (sunkickr) [reply cloud/deployment/deployment_test.go:1289]: added comments

**Review by kushalmalani — APPROVED** [review]

**Pull request merged**